apps/arena/lib/arena/game/skill.ex

```diff
@@ -232,15 +257,101 @@ defmodule Arena.Game.Skill do
 
       Process.send_after(self(), {:remove_effect, player.id, last_id}, effect.duration_ms)
 
-      effects =
-        get_in(game_state, [:players, player.id, :aditional_info, :effects])
-        |> Map.put(last_id, effect)
-
-      put_in(game_state, [:players, player.id, :aditional_info, :effects], effects)
+      put_in(
+        game_state,
+        [:players, player.id, :aditional_info, :effects, last_id],
+        Map.put(effect, :id, last_id)
+      )
       |> put_in([:last_id], last_id)
     end)
   end
 
+  def apply_effect_mechanic(players, game_state) do
+    updated_players =
+      Map.filter(players, fn {_, player} -> Player.alive?(player) end)
+      |> Map.new(fn {_player_id, player} ->
+        player =
+          Enum.reduce(player.aditional_info.effects, player, fn {_effect_id, effect}, player ->
+            apply_effect_mechanic(player, effect, game_state)
+          end)
+
+        {player.id, player}
+      end)
+
+    Map.merge(players, updated_players)
+  end
+
+  def apply_effect_mechanic(player, effect, game_state) do
+    now = DateTime.utc_now() |> DateTime.to_unix(:millisecond)
```

Comment:
Can we use `System.monotonic_time/2`?

Reply:
hmmm is there any benefit of using System.monotonic_time over the utc now?

Reply:
Its way faster, a quick benchmark in my laptop

```elixir
Benchee.run(
  %{
    "DateTime.to_unix/2" => fn -> DateTime.utc_now() |> DateTime.to_unix(:millisecond) end,
    "System.monotonic_time/1" => fn -> System.monotonic_time(:millisecond) end
  }
)
```

Results
```
Name                              ips        average  deviation         median         99th %
System.monotonic_time/1       27.56 M       36.28 ns  ±7595.42%       33.40 ns       45.90 ns
DateTime.to_unix/2             2.81 M      356.43 ns  ±8331.29%         292 ns         459 ns

Comparison: 
System.monotonic_time/1       27.56 M
DateTime.to_unix/2             2.81 M - 9.82x slower +320.14 ns
```

Reply:
wow! speaking about upgrades. Should be fixed now, also i've noticed that i was adding the last application time to the effect instead of  to each specific mechanic when i was adding the effects to the players.